### PR TITLE
Fixed issue  #22409 Configurable products are shown on Select Associated Product modal

### PR DESCRIPTION
**Pull request by mallikaverma**
Fixed issue  #22409 Configurable products are shown on Select Associated Product modal

### Description (*)
Fixed issue  #22409 Configurable products are shown on Select Associated Product modal

### Fixed Issues (if relevant)
Fixed issue  #22409 Configurable products are shown on Select Associated Product modal

1. magento/magento2#22409: Configurable products are shown on Select Associated Product modal

### Manual testing scenarios (*)
1.Create a configurable product Test 1 with 2 child products (color = red and color = black for the example)
2. Create a new simple product Test 2 with color = red
3. Open the last created product Test 2 in admin panel and convert it to configurable by adding child products (color = green, color = grey), do not change the value for color attribute while you edit product
4. Open the first configurable product Test 1, remove the child product with color = red, click on 'Add products manually' link in Configurations section

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+ * @api
+ * @since 100.0.2
+ */
+class AssociatedDataProvider extends \Magento\Catalog\Ui\DataProvider\Product\ProductDataProvider
+{
+
+    /**
+     * @var Magento\Catalog\Model\ProductFactory
+     */
+    protected $productData;
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:27]: should be private

also the naming seems off (product data vs. product factory)

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    /**
+     * Product collection
+     *
+     * @var \Magento\Catalog\Model\ResourceModel\Product\Collection
+     */
+    protected $collection;
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:34]: already exists in the parent class
There seems to be no reason to override it here

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+    /**
+     * @var \Magento\Ui\DataProvider\AddFieldToCollectionInterface[]
+     */
+    protected $addFieldStrategies;
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:39]: already exists in the parent class
There seems to be no reason to override it here

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+    /**
+     * @var \Magento\Ui\DataProvider\AddFilterToCollectionInterface[]
+     */
+    protected $addFilterStrategies;
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:44]: already exists in the parent class
There seems to be no reason to override it here

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    /**
+     * @var StoreInterface
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:47]: import for StoreInterface is missing (getting undefined class warning)

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        ProductFactory $productFactory,
+        LocatorInterface $locator,
+        StoreRepositoryInterface $storeRepository,
+        RequestInterface $request,
+        array $meta = [],
+        array $data = [],
+        array $addFieldStrategies = [],
+        array $addFilterStrategies = []
+    ) {
+        $this->_configurableType = $configurableType;
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:83]: this class property does dot exist (field declared dynamically warning)

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        $this->productData = $productFactory;
+        $this->locator = $locator;
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:85]: this class property does dot exist (field declared dynamically warning)

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+        $this->request = $request;
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:86]: this class property does dot exist (field declared dynamically warning)

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+        $this->storeRepository = $storeRepository;
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:87]: this class property does dot exist (field declared dynamically warning)

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+
+        return $collection;
+    }
+
+    /**
+     * Get product
+     *
+     * @return \Magento\Catalog\Model\Product
+     */
+    protected function _getProduct()
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:138]: method should be marked private (see [health index details ](https://testing-service.magento-community.engineering/reports/magento/magento2/pull/22725/44f04ea5-239e-437f-858c-906c984910ff/4115/MagentoHealthIndex/report.html)check) and not prefixed with an underscore

File: app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+        return $this->productData->create()->load($product_id);
+    }
+
+    /**
+     * Retrieve store
+     *
+     * @return StoreInterface|\Magento\Store\Api\Data\StoreInterface|null
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     */
+    protected function getStore()
```
Comment (davidverholen) [inline app/code/Magento/ConfigurableProduct/Ui/DataProvider/Product/Associated/AssociatedDataProvider.php:150]: method should be marked private
(https://testing-service.magento-community.engineering/reports/magento/magento2/pull/22725/44f04ea5-239e-437f-858c-906c984910ff/4115/MagentoHealthIndex/report.html)

File: app/code/Magento/Catalog/Model/Product/Type.php [hunk 12]
```diff
@@ -24,6 +24,9 @@ class Type implements OptionSourceInterface
     const TYPE_BUNDLE = 'bundle';
```
Comment (davidverholen) [inline app/code/Magento/Catalog/Model/Product/Type.php:24]: these type constants do already exist and should be reused from there
for example: \Magento\Bundle\Model\Product\Type::TYPE_CODE

**Review by davidverholen — CHANGES_REQUESTED** [review]

**Comment by sidolov** [comment]
@mallikaverma , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**